You have been tasked with answering three key questions about New York City (NYC) public school SAT performance.

In [ ]:
# Re-run this cell 
import pandas as pd

# Read in the data
schools = pd.read_csv("schools.csv")

# Preview the data
schools.head()



school_name  ... percent_tested
0  New Explorations into Science, Technology and ...  ...            NaN
1                               Essex Street Academy  ...           78.9
2                       Lower Manhattan Arts Academy  ...           65.1
3    High School for Dual Language and Asian Studies  ...           95.9
4      Henry Street School for International Studies  ...           59.7

[5 rows x 7 columns]

# Question 1

### **Which NYC schools have the best math results?**

The best math results are defined as scores in the `average_math` column that are greater than or equal to 80% of 800.

We will filter out the rows that return true for this condition using loc and select the `school_name` and `average_math` columns

In [238]:
best_math_schools = schools.loc[schools['average_math'] >= 0.8 * 800][['school_name','average_math']].sort_values(by="average_math",ascending=False)
best_math_schools

school_name  average_math
88                              Stuyvesant High School           754
170                       Bronx High School of Science           714
93                 Staten Island Technical High School           711
365  Queens High School for the Sciences at York Co...           701
68   High School for Mathematics, Science, and Engi...           683
280                     Brooklyn Technical High School           682
333                        Townsend Harris High School           680
174  High School of American Studies at Lehman College           669
0    New Explorations into Science, Technology and ...           657
45                       Eleanor Roosevelt High School           641

We find that there are 10 schools that have the 'best' math results

# Question 2

### What are the top 10 performing schools based on the combined SAT scores?

To calculate the combined SAT scores, we will sum the values in the `average_math`, `average_reading`, and `average_writing` columns. The result of this calculation will be stored in a new column named `total_SAT`.

To identify the **top 10 schools**, we will sort the `schools` DataFrame in descending order based on the `total_SAT` column and then filter to obtain the top 10 rows using `iloc`.

In [239]:
schools['total_SAT'] = schools['average_math'] + schools['average_writing'] + schools['average_reading']
top_10_schools = schools[['school_name', 'total_SAT']].sort_values(by="total_SAT", ascending=False).iloc[0:10]
top_10_schools

school_name  total_SAT
88                              Stuyvesant High School       2144
170                       Bronx High School of Science       2041
93                 Staten Island Technical High School       2041
174  High School of American Studies at Lehman College       2013
333                        Townsend Harris High School       1981
365  Queens High School for the Sciences at York Co...       1947
5                       Bard High School Early College       1914
280                     Brooklyn Technical High School       1896
45                       Eleanor Roosevelt High School       1889
68   High School for Mathematics, Science, and Engi...       1889

The top 10 performing schools have total SAT scores between 2144 and 1889

# Question 3

### Which single borough has the largest standard deviation in the combined SAT score?

We first need to group all rows in the `schools` dataframe by borough and apply the `std` function

In [240]:
largest_std_dev = schools.groupby(['borough']).std()
largest_std_dev = round(largest_std_dev, 2).sort_values(by="total_SAT", ascending=False).iloc[0:1]


We can see that the Manhattan `borough` has the largest standard deviation

Now we add calculate the number of schools in Manhattan and the mean total SAT score, and add those values as columns

In [241]:
total_num_schools = (schools['borough'] == 'Manhattan').sum()
largest_std_dev['num_schools'] = round(total_num_schools,2)

total_average_SAT = schools[schools['borough'] == 'Manhattan']['total_SAT'].mean()
largest_std_dev['average_SAT'] = round(total_average_SAT,2)


To finalize we rename the `total_SAT` column as `std_SAT` and extract the relevant columns for our new dataframe

In [242]:
largest_std_dev.rename(columns={'total_SAT':'std_SAT'}, inplace=True)
largest_std_dev = largest_std_dev[['num_schools','average_SAT','std_SAT']]

largest_std_dev

num_schools  average_SAT  std_SAT
borough                                     
Manhattan           89      1340.13   230.29